rpcserver.go

```diff
@@ -1185,18 +1186,66 @@ func prepareAndSubmitOrder(ctx context.Context, o order.Order,
 	)
 }
 
+// lndVersionIsCompatible makes sure the detected lnd version is compatible
+// with our current version requirements. True is returned if the version is
+// compatible, and false otherwise.
+func lndVersionIsCompatible(actual *verrpc.Version,
```

Comment:
I have some dobuts here:

- Do we need to check something else beyond the lnd version (`featurebits`?)

- Do we need to check the `BuildTags`?

- May be worth moving this function to `"github.com/lightningnetwork/lnd/lnrpc/verrpc"` itself so we do not have to copy paste it in multiple projects.

- Here we assume that all versions will be backwards compatible, we need to keep this in mind in the future then.

Reply:
It's possible the user has lnd `v0.14.x` but explicitly disables the script enforcement functionality. But then they would just reject any matched orders in the channel acceptor.

We don't need to check build tags though. They are checked on startup already. So we could leave them empty in the `scriptEnforceVersion` variable.

I created a PR for exporting `lndclient.AssertVersionCompatible`, you just need to bump the `lndclient` version to `v0.14.0-7` to get it.

Reply:
Thanks for that PR @guggero! Using that function directly now instead of copying things over.